functions/v2/cloud-event-logging/test/index.test.js

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+const {getFunction} = require('@google-cloud/functions-framework/testing');
+const {CloudEvent} = require('cloudevents');
+const pkg = require('../package.json');
+
+describe('structured logging: functions cloudevent', () => {
+  before(async () => {
+    await import("../index.js");
+    // Adding a spy to the write function in stdout
+    sinon.spy(process.stdout, 'write');
+  });
+
+  afterEach(() => {
+    process.stdout.write.restore();
+  });
+
+  it('structuredLoggingEvent: should correctly print logs', async () => {
+    // Declare in-line when you run the test (i.e GOOGLE_CLOUD_PROJECT=abc npm run test)
+    const { TOPIC_NAME, GOOGLE_CLOUD_PROJECT } = process.env;
+    const structuredLoggingEvent = getFunction('structuredLoggingEvent');
+
+    const expected = {
+      severity: 'NOTICE',
+      component: 'arbitrary-property',
+      logName: `projects/${GOOGLE_CLOUD_PROJECT}/logs/${pkg.name}`,
+      message: 'Hello, world!'
+    };
+
+    // Passing in a mocked CloudEvent object
+    await structuredLoggingEvent(new CloudEvent({
+      type: 'google.cloud.pubsub.topic.v2.messagePublished',
+      source:`//pubsub.googleapis.com/${TOPIC_NAME}`,
+    }));
+
+    // Capture the string output & parse
+    const result = JSON.parse(process.stdout.write.getCall(0).args[0]);
```

Comment:
Do we want to assert the following, or is this overkill? (Those "step by step" details can be helpful when debugging on CI.)
* `JSON.parse` of `process.stdout.write.getCall(0).args[0]` succeeds
* `process.stdout.write` was called (at least) once
* `process.stdout.write.getCall(0).args` is non-empty

Reply:
Sure, I can add a few more assertions